app/app.go

```diff
@@ -399,6 +414,13 @@ func NewGaiaApp(
 		),
 	)
 	app.SetEndBlocker(app.EndBlocker)
+	app.UpgradeKeeper.SetUpgradeHandler("Gravity-DEX",
```

Comment:
Since this will be a _new_ module added to an _existing_ chain in a _live_ upgrade, do we need to anything special for telling the upgrade handler that this is a new module? I recall hearing issues/bugs where queries and other things failed because _new_ modules were added in upgrades. Maybe this is no longer an issue -- just wanted to be sure.

ref: https://github.com/cosmos/cosmos-sdk/pull/9526

Reply:
Yes, we need to tell the multi-store about the new store for the liquidity module. This will go away once we migrate away from the multi-store altogether. But for now this step is needed.

See the docs here: https://pkg.go.dev/github.com/cosmos/cosmos-sdk@v0.42.5/x/upgrade#hdr-Performing_Upgrades

I'll suggest the code below

Reply:
Unfortunately this store upgrade cannot happen in the upgrade handler because the multi-store loads the stores _before_ the app starts. So we need to read the upgrade info from disk, upgrade the multi-store, then when the app starts the upgrade handler above gets called during begin block to initialize the liquidity module's genesis. Hope that's clear.